trial_balance.py

```diff
@@ -439,11 +440,11 @@ def _record(account, party, vals, comp, add_initial_balance):
                             fiscalyear.company)
 
                 logger.info('Calc values for comparsion for parties')
-                with transaction.set_context(fiscalyear=fiscalyear.id,
+                with transaction.set_context(fiscalyear=comparison_fiscalyear.id,
                         periods=comparison_periods):
                     comparison_party_values = \
                         Party.get_account_values_by_party(parties, accounts,
```

Comment:
LGTM

get_account_values_by_party() call MoveLine.query_get(line) and get fiscalyer or periods from the context [1]

[1]  https://github.com/tryton/account/blob/develop/move.py#L982